### ISSUE TITLE
Prediction error plot uses the same color for the train and test

### ISSUE THREAD

**Issue by yoavram** [DistrictDataLabs/yellowbrick#1100]
I would expect the prediction error visualiser to plot the train and test sets with different colors, but it uses the same colors.

```python
import sklearn.datasets
from sklearn.neighbors import KNeighborsRegressor
from sklearn.model_selection import train_test_split
from yellowbrick.regressor import prediction_error

X, y = sklearn.datasets.load_diabetes(return_X_y=True)
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.25, random_state=0)
regressor = KNeighborsRegressor()
regressor.fit(X_train, y_train)
prediction_error(regressor, X_train, y_train, X_test, y_test, )
```

![](https://i.postimg.cc/s2jgq3WD/attachment.png)

I would expect to see blue and red (or any other two colors) markers for the train and test, but there is only one color for both the train and the test.

 - OS: macOS
 - Python Version 3.8
 - Yellowbrick Version 1.1

**Comment by bbengfort** [comment]
@yoavram thank you for using Yellowbrick and for your question! In the example figure you've provided above, _you're only seeing the test data_, not both the training and test data. The `PredictionError` visualizer works similarly to a scikit-learn scoring model, where you would first fit the model on `X_train` and `y_train` then get the score on `X_test` and `y_test`. What `PredictionError` is trying to visualize is a better indication of what the `R2` score value is, in terms of variance and heteroskedasticity in the predictions. 

I absolutely understand the use case where you might want to compare model performance on training data vs test data in order to attempt to understand if the model is overfitted. For this, I would strongly recommend the [`ResidualsPlot`](https://www.scikit-yb.org/en/latest/api/regressor/residuals.html), which has additional utilities for this type of analysis. However, there are two ways to make this type of analysis work with `PredictionError`:

First, you can simply draw your `PredictionError` plots twice on the same Axes object, using different colors for the train and test set. Since you've already fitted your visualizer prior to drawing the plots, this will simply score the model on both the training and the test sets:

```python
# Create a single axes to draw both plots on 
_, ax = plt.subplots()

# Draw the training data prediction error plot
prediction_error(
    regressor, X_train, y_train, 
    ax=ax, is_fitted=True, show=False, 
    point_color='b', line_color='b', 
)

# Draw the test data prediction error plot
prediction_error(
    regressor, X_test, y_test, 
    ax=ax, is_fitted=True, show=False, 
    point_color='g', line_color='g', 
)

# Note that we used show=False above, if you're not in a notebook, you must show the figure now. 
plt.show()
```

![ontop](https://user-images.githubusercontent.com/745966/94946911-cde1a280-04aa-11eb-9d50-f87a0673f369.png)

This to me is less than ideal since it is completely crowded and doesn't really allow you to do the type of visual analysis `PredictionError` was intended for. Instead, I recommend putting both the training and test data in separate plots in the same figure as follows:

```python
# Create a figure with two axes objects
fig, axes = plt.subplots(ncols=2)

# Draw the training data prediction error plot
prediction_error(
    regressor, X_train, y_train, 
    ax=axes[0], is_fitted=True, show=False, 
    point_color='b', line_color='b', 
    title="Training Data",
)

# Draw the test data prediction error plot
prediction_error(
    regressor, X_test, y_test, 
    ax=axes[1], is_fitted=True, show=False, 
    point_color='g', line_color='g', 
    title="Test Data",
)

# Show the figure - note that we passed show=False above to do a multi-axes figure
fig.tight_layout()
plt.show()
```

![side](https://user-images.githubusercontent.com/745966/94947219-45173680-04ab-11eb-82f8-ddd04d91cf9d.png)

Note that both of these methods rely on passing `show=False` to the quick method, this isn't necessary when using the Visualizer directly. Additionally, this method also relies on using a fitted estimator, but your original example also did this. 

In the second example, compared to the first, it is much easier to identify the variance of predictability and even though it's harder to see the relative slopes of the best fit lines, I think it's much clearer what is happening in the performance of the model on the training data and test data.
